src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1241,4 +1245,38 @@ public static void onEntityEnterSection(Entity entity, long packedOldPos, long p
         MinecraftForge.EVENT_BUS.post(new EntityEvent.EnteringSection(entity, packedOldPos, packedNewPos));
     }
 
+    /**  FOR INTERNAL USE ONLY, DO NOT CALL DIRECTLY */
+    public static void injectRegistryAccessExtensions(Map<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> registryData, Map<ResourceKey<? extends Registry<?>>, MappedRegistry<?>> builtin)
+    {
+        for (var extension : ForgeRegistries.REGISTRY_ACCESS_EXTENSIONS)
+        {
+            var key = extension.getRegistryKey();
+            var data = createExtensionData(extension);
+            var registry = copyExtensionRegistry(extension);
+
+            // Injects the ResourceKey and Codec to the RegistryAccess.REGISTRIES map
+            registryData.put(key, data);
+
+            // Injects the mod-provided built-in registry elements to RegistryAccess.BUILTIN
+            builtin.put(key, registry);
+
+            LOGGER.info("Injected RegistryAccessExtension: {}", key);
+        }
+    }
+
+    private static <E extends IForgeRegistryEntry<E>> RegistryAccess.RegistryData<E> createExtensionData(RegistryAccessExtension<E> extension)
+    {
+        return new RegistryAccess.RegistryData<>(extension.getRegistryKey(), extension.getDirectCodec(), null);
```

Comment:
So are network codecs not supported by this as well and should be left as a feature to implement of the modders? If so, that should be specified somewhere.

Reply:
It'd be easy to add (an extra codec field in RegAccExtn and using that here instead of the null). 

The consideration to make is that the client needs the same set of codecs in order to connect to a server that sends extra components (see attached). There's no facility for ensuring that at the mo besides implying it by synchronizing the builtin registry for those components (still doesn't guarantee a codec exists on the client). I feel like if there's a strong argument for sync'ing it could be looked at in separate PR. The usefullness of this system imo lies entirely on the server side

_This is a vanilla client connecting to a server that is sending extra world-gen components over the network:_
![2022-01-06](https://user-images.githubusercontent.com/71294196/148543478-f991e597-08d9-460c-bfbc-ed11836e7384.png)